app/views/unattended/provisioning_templates/snippet/first_boot_setup.erb

```diff
@@ -0,0 +1,55 @@
+<%#
+kind: snippet
+name: first_boot_setup
+model: ProvisioningTemplate
+snippet: true
+description: |
+  Post replacement for both systemd and non-systemd platforms
+-%>
+<%
+  os_major = @host.operatingsystem.major.to_i
+  rhel_compatible = @host.operatingsystem.family == 'Redhat' && @host.operatingsystem.name != 'Fedora'
+  has_systemd = (@host.operatingsystem.name == 'Fedora' && os_major >= 20) || (rhel_compatible && os_major >= 7) || (@host.operatingsystem.name == 'Ubuntu' && os_major >= 15) || (@host.operatingsystem.name == 'Debian' && os_major >= 8)
+  initd_script = """
```

Comment:
We have a special snippet for the service, maybe would be worthy to have a snippet for the `initd_script` as well